private/bufpkg/bufmodule/added_module.go

```diff
@@ -160,6 +161,28 @@ func (a *addedModule) ToModule(
 		}
 		return moduleData.V1Beta1OrV1BufLockObjectData()
 	}
+	getB5DigestForRemoteModule := func() (Digest, error) {
+		moduleKeyDigest, err := a.remoteModuleKey.Digest()
+		if err != nil {
+			return nil, err
+		}
+		if moduleKeyDigest.Type() == DigestTypeB5 {
+			return moduleKeyDigest, nil
+		}
+		// Our module key was created with a b4 digest - we have to look up the b5 digest for the commit.
+		commitKey, err := NewCommitKey(a.remoteModuleKey.ModuleFullName().Registry(), a.remoteModuleKey.CommitID(), DigestTypeB5)
```

Comment:
This is the kinda thing that would be good to cache so we don't have to look it up on the fly when we're dealing with b4 module keys.